use-cases/virgo/data.py

```diff
@@ -61,159 +67,158 @@ def execute(self) -> pd.DataFrame:
-        #     for i in range(100)
-        # ])  # for i in range(y_train.shape[0])
-        # aux_data_train_small_2d = torch.stack([
-        #     torch.stack([X_train_2d.iloc[i][0], X_train_2d.iloc[i]
-        #                 [1], X_train_2d.iloc[i][2]])
-        #     for i in range(100)
-        # ])  # for i in range(X_train.shape[0])
-
-        # whole dataset
-        signal_data_train_2d = torch.stack([
-            torch.stack([y_train_2d[main_channel].iloc[i]])
-            for i in range(y_train_2d.shape[0])
-        ])
-        aux_data_train_2d = torch.stack([
-            torch.stack(
-                [X_train_2d.iloc[i][0], X_train_2d.iloc[i][1],
-                 X_train_2d.iloc[i][2]])
-            for i in range(X_train_2d.shape[0])
-        ])
-
-        # concatenate torch.tensors
-        train_data_2d = torch.cat(
-            [signal_data_train_2d, aux_data_train_2d], dim=1)
-        # train_data_small_2d = torch.cat(
-        #     [signal_data_train_small_2d, aux_data_train_small_2d], dim=1)
-
-        # VALIDATION SET
-
-        # # smaller dataset
-        # signal_data_test_small_2d = torch.stack([
-        #     torch.stack(
-        #         [y_test_2d[main_channel].iloc[i]])
-        #     for i in range(100)
-        # ])  # for i in range(y_test.shape[0])
-        # aux_data_test_small_2d = torch.stack([
-        #     torch.stack(
-        #         [X_test_2d.iloc[i][0], X_test_2d.iloc[i][1],
-        #          X_test_2d.iloc[i][2]])
-        #     for i in range(100)
-        # ])  # for i in range(X_test.shape[0])
-
-        # whole dataset
-        signal_data_test_2d = torch.stack([
-            torch.stack(
-                [y_test_2d[main_channel].iloc[i]])
-            for i in range(y_test_2d.shape[0])
-        ])
-        aux_data_test_2d = torch.stack([
-            torch.stack(
-                [X_test_2d.iloc[i][0], X_test_2d.iloc[i][1],
-                 X_test_2d.iloc[i][2]])
-            for i in range(X_test_2d.shape[0])
-        ])
-
-        test_data_2d = torch.cat(
-            [signal_data_test_2d, aux_data_test_2d], dim=1)
-        # test_data_small_2d = torch.cat(
-        #     [signal_data_test_small_2d, aux_data_test_small_2d], dim=1)
-
-        # NORMALIZE
-        train_data_2d = normalize_(train_data_2d)
-        test_data_2d = normalize_(test_data_2d)
-
-        return train_data_2d, test_data_2d, None
+
+        # Ensure the root folder exists
+        if not os.path.isdir(self.root_folder):
+            raise FileNotFoundError(f"Root folder '{self.root_folder}' not found.")
+
+        # Find all file paths in root folder
+        all_file_paths = [os.path.join(dirpath, f)
+                          for dirpath, dirs, files in os.walk(self.root_folder)
+                          for f in files if f.endswith('.pkl')]
+
+        # Split file paths into train, validation, and test sets
+        [train_paths, validation_paths, test_paths] = random_split(
```

Comment:
I don't think you need the square brackets when unpacking these variables. You could probably just do 
train_paths, validation_paths, test_paths = random_split(...)